__tests__/app.test.js

```diff
@@ -40,8 +40,38 @@ describe("GET /api", () => {
       .expect(200)
       .then((response) => {
         const endpoints = require("../endpoints.json");
-        const endpoint = response.body.endpoints 
+        const endpoint = response.body.endpoints;
         expect(endpoints).toEqual(endpoint);
       });
   });
 });
+
+describe("GET /api/articles/:article_id", () => {
+  test("GET:200 status responds with the correct article object", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .expect(200)
+      .then((res) => {
+        const article = res.body.articles;
+        expect(article.article_id).toBe(2);
+        expect(article.title).toBe("Sony Vaio; or, The Laptop");
+      });
+  });
+  test("GET:404 sends an appropriate status and error message when given a valid but non-existent id", () => {
+    return request(app)
+      .get("/api/articles?article_id=9999")
```

Comment:
You are getting a 404 but you are not handling this error in the way you possibly should be. 

You have set up a parametric endpoint in your app.js - articles/:article_id

However in your test here you are never hitting that endpoint because you are using query parameters in your test - ?article_id-9999

This will actually be hitting the middleware you have to catch invalid endpoints. 

What you should be doing is changing the test to be the following - /api/articles/9999 and in your model when you get an empty array back from the database - you should be handling that with a Promise.reject() which will pass the error onto your error handling middleware.